src/Core/Framework/DataAbstractionLayer/Dbal/FieldResolver/ManyToOneAssociationFieldResolver.php

```diff
@@ -124,21 +121,50 @@ protected function getSourceColumn(EntityDefinition $definition, AssociationFiel
     }
 
     /**
-     * @internal Overwritten in parent association field resolver to add join conditions for inherited associations
+     * Builds a versioning string to append in the where condition based on the available fields
+     *
+     * @param ManyToOneAssociationField|OneToOneAssociationField $field
      */
-    protected function buildWhere(AssociationField $field, FieldResolverContext $context): string
+    private function buildVersionWhere(FieldResolverContext $context, AssociationField $field): string
     {
-        $reference = $field->getReferenceDefinition();
+        if (!$field->getReferenceDefinition()->isVersionAware()) {
+            return '';
+        }
 
-        $versionAware = $context->getDefinition()->isVersionAware() && $reference->isVersionAware();
+        $fkVersionId = $field->getReferenceDefinition()->getEntityName() . '_version_id';
```

Comment:
sorry if I bring this up again now, but somehow the code doesn't make sense to me...

A many-to-one association is for me always a linking of data. That means the data does not "belong" directly to the linked entity.

Therefore I think that the data should not be loaded in the "same" version but always in the live version.

You have brought the example `document` <> `order` in the ticket. So you load the document here and want to load the order to it do I see that right?

Shouldn't there be a check if the entities "belong together"? I.e. you would have to check if the definition that is joined here is the same as the `getParentDefinitionClass` to make sure that it is the same entity versioning and not a completely different one.

Because if I want to load the product from the order line item, for example, the changes would try to load the product with the version ID of the order correct? And this seems not correct for me.